dashboard/app/views/layouts/_header.html.haml

```diff
@@ -53,9 +53,15 @@
               = render partial: 'shared/user_header'
 
 - if script_level
-  - user_progress = current_user == nil ? 'null' : summarize_user_progress(script_level.script, @user || current_user).to_json
 
   :javascript
     //<![CDATA[
-    dashboard.buildHeader(#{script_level.stage.summarize.to_json}, #{user_progress}, #{script_level.level_id}, #{@user.try(:id) || 'null'}, #{@section.try(:id) || 'null'})
+    var buildHeader = function () {
+        dashboard.buildHeader(#{script_level.stage.summarize.to_json}, #{script_level.level_id}, #{@user.try(:id) || 'null'}, #{@section.try(:id) || 'null'});
+    };
+    if(#{!!(@current_user).to_json}) {
+      dashboard.clientState.refreshScriptProgressFromServer(#{script_level.script_id}).always(buildHeader);
+    } else {
+      buildHeader();
+    }
```

Comment:
The server response might never come, so it would be be better render something initially and then re-render when/if the server response came back.